### ISSUE TITLE
Installation issue: gettext

### ISSUE THREAD

**Issue by etiennemlb** [LLNL/spack#46863]
### Steps to reproduce the issue

```console
$ spack spec -I gettext@0.22.5%gcc@13.2.0~libxml2
$ spack spec -I gettext@0.22.5%gcc@13.2.0+bzip2+curses+git~libunistring~libxml2+pic+shared+tar+xz
Input spec
--------------------------------
 -   gettext@0.22.5%gcc@13.2.0+bzip2+curses+git~libunistring~libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rhel8-zen3

Concretized
--------------------------------
 -   gettext@0.22.5%gcc@13.2.0+bzip2+curses+git~libunistring~libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rhel8-zen3
[+]      ^bzip2@1.0.8%gcc@13.2.0~debug~pic+shared build_system=generic arch=linux-rhel8-zen3
[+]          ^diffutils@3.10%gcc@13.2.0 build_system=autotools arch=linux-rhel8-zen3
[+]      ^gcc-runtime@13.2.0%gcc@13.2.0 build_system=generic arch=linux-rhel8-zen3
[e]      ^glibc@2.28%gcc@13.2.0 build_system=autotools arch=linux-rhel8-zen3
[+]      ^gmake@4.4.1%gcc@13.2.0~guile build_system=generic arch=linux-rhel8-zen3
[+]      ^libiconv@1.17%gcc@13.2.0 build_system=autotools libs=shared,static arch=linux-rhel8-zen3
[+]      ^ncurses@6.5%gcc@13.2.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-rhel8-zen3
[+]          ^pkgconf@2.2.0%gcc@13.2.0 build_system=autotools arch=linux-rhel8-zen3
[+]      ^tar@1.34%gcc@13.2.0 build_system=autotools zip=pigz arch=linux-rhel8-zen3
[+]          ^pigz@2.8%gcc@13.2.0 build_system=makefile arch=linux-rhel8-zen3
[+]              ^zlib-ng@2.1.6%gcc@13.2.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-rhel8-zen3
[+]          ^zstd@1.5.6%gcc@13.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-rhel8-zen3
[+]      ^xz@5.4.6%gcc@13.2.0~pic build_system=autotools libs=shared,static arch=linux-rhel8-zen3
```

Only occurs when deactivating libxml2 variant.

@michaelkuhn

### Error message

<details><summary>Error message</summary>

<pre>
==> [2024-10-08-12:12:12.892254] Error: ProcessError: Command exited with status 2:
    'make' '-j24' 'V=1'

6 errors found in build log:
     8020    <redacted>/System/Externals/spack/lib/spack/env/gcc/gcc: line 963: <redacted>/spack-cc
             -gettext-hnmf722.in.log: Read-only file system
     8021    <redacted>/System/Externals/spack/lib/spack/env/gcc/gcc: line 965: <redacted>/spack-cc
             -gettext-hnmf722.out.log: Read-only file system
     8022    <redacted>/System/Externals/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE
                 -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I<redacted>/Environments/Base/Binaries/linux-rhel8-zen3/gcc-13.2.0/libiconv-
             1.17-73lklohbhbksuhhjjnu7on65hcpzfbv5/include  -g -O2 -w -fno-analyzer -c term-styled-ostream.c
     8023    <redacted>/System/Externals/spack/lib/spack/env/gcc/gcc: line 963: <redacted>/spack-cc
             -gettext-hnmf722.in.log: Read-only file system
     8024    <redacted>/System/Externals/spack/lib/spack/env/gcc/gcc: line 965: <redacted>/spack-cc
             -gettext-hnmf722.out.log: Read-only file system
     8025    In file included from term-styled-ostream.oo.c:26:
  >> 8026    ./libcroco/cr-sel-eng.h:43:10: fatal error: libxml/tree.h: No such file or directory
     8027       43 | #include <libxml/tree.h>
     8028          |          ^~~~~~~~~~~~~~~
     8029    compilation terminated.
  >> 8030    make[4]: *** [Makefile:6192: config.h] Error 1
     8031    make[4]: Leaving directory '/tmp<redacted>/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src/libtextstyle/lib'
  >> 8032    make[3]: *** [Makefile:1999: all-recursive] Error 1
     8033    make[3]: Leaving directory '/tmp<redacted>/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src/libtextstyle'
  >> 8034    make[2]: *** [Makefile:1927: all] Error 2
     8035    make[2]: Leaving directory '/tmp<redacted>/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src/libtextstyle'
  >> 8036    make[1]: *** [Makefile:408: all-recursive] Error 1
     8037    make[1]: Leaving directory '/tmp<redacted>/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src'
  >> 8038    make: *** [Makefile:364: all] Error 2
</pre></details>

### Information on your system

$ spack debug report
* **Spack:** 0.22.2 (594a376c521cc746978571b1181a47bbcff30a21)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen4
* **Concretizer:** clingo

### Additional information

[logs.tar.gz](https://github.com/user-attachments/files/17292206/logs.tar.gz)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by yizeyi18** [comment]
Here gettext uses its included libxml, the error may come from this. 
May you check `tree.h` in your `/tmp/malaboeuf/spack-stage/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src`, by `find /tmp/malaboeuf/spack-stage/spack-stage-gettext-0.22.5-hnmf7226pqybdymdpmo3wnvlfihw3527/spack-src -name "tree.h"`, to make sure whether the file exists or not?

**Comment by bernhardkaindl** [comment]
@etiennemlb, @yizeyi18: This is fixed by #46870
